test/e2e/opaque_resource.go

```diff
@@ -0,0 +1,309 @@
+		awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+			rQuant, found := n.Status.Capacity[opaqueResName]
+			return found && rQuant.MilliValue() == int64(5000)
+		})
+
+		By("Watching node status to verify the advertised opaque resource is eventually present in allocatable")
+		awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+			rQuant := n.Status.Allocatable[opaqueResName]
+			return rQuant.MilliValue() == int64(5000)
+		})
+	})
+
+	Context("With opaque resources already advertised", func() {
+		BeforeEach(func() {
+			node = refresh(f, node)
+			// Use the client to patch node capacity, adding opaque "foo" resource.
+			patch := []byte(`[{"op": "add", "path": "/status/capacity/pod.alpha.kubernetes.io~1opaque-int-resource-foo", "value": "5"}]`)
+			f.Client.Patch(api.JSONPatchType).Resource("nodes").Name(node.Name).SubResource("status").Body(patch).Do()
+			awaitNodePredicate(f.Client, node, func(n *api.Node) bool {
+				rQuant, found := n.Status.Allocatable[opaqueResName]
+				return found && rQuant.MilliValue() == int64(5000)
+			})
+			node = refresh(f, node)
+		})
+
+		It("should not break pods that do not consume opaque integer resources.", func() {
+			By("Verifying that a vanilla pod can still be scheduled")
+			requests := api.ResourceList{api.ResourceCPU: resource.MustParse("0.1")}
+			limits := api.ResourceList{api.ResourceCPU: resource.MustParse("0.2")}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "without-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pod).To(beScheduled())
+		})
+
+		It("should schedule pods that do consume opaque integer resources.", func() {
+			By("Creating a pod that requires less of the opaque resource than is allocatable on a node.")
+			requests := api.ResourceList{
+				api.ResourceCPU: resource.MustParse("0.1"),
+				opaqueResName:   resource.MustParse("1"),
+			}
+			limits := api.ResourceList{
+				api.ResourceCPU: resource.MustParse("0.2"),
+				opaqueResName:   resource.MustParse("2"),
+			}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "min-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(pod).To(beScheduled())
+		})
+
+		It("should not schedule pods that exceed the available amount of opaque integer resource.", func() {
+			By("Creating a pod that requires more of the opaque resource than is allocatable on any node")
+			requests := api.ResourceList{opaqueResName: resource.MustParse("6")}
+			limits := api.ResourceList{}
+			pod, err := f.Client.Pods(f.Namespace.Name).Create(newTestPod(f, "over-max-oir", requests, limits))
+			Expect(err).NotTo(HaveOccurred())
+			// TODO(CD): Watch scheduler events instead to catch an indication of
+			//           insufficient resources or other unschedulable condition
+			//           instead of waiting for this to time out.
+			Expect(pod).To(notBeScheduled())
```

Comment:
There are other tests that do this per-similar request: 

see: https://github.com/kubernetes/kubernetes/pull/26501/files#diff-331b6cbc690d4629a538fe6b8d293bc1R69